### PR TITLE
JAMES-1872: Add integration test for JMAP for messages using two times the same CID

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java [hunk 1]
```diff
@@ -235,7 +240,7 @@ public void postWithParameters(List<String> ids, List<String> properties) throws
                 .collect(Collectors.joining(",", "[", "]" ));
 
         String serializedProperties = properties.stream()
-                .map(toJsonString())
+            .map(toJsonString())
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: You broke indent

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+--=_ebeed74b39dc10b43908b65b893b4ea4
+Content-Transfer-Encoding: base64
+Content-ID: <1482981567586480bfca67b793175279@linagora.com>
+Content-Type: image/png;
+ name="g3619(1).png"
+Content-Disposition: inline;
+ filename="g3619(1).png";
+ size=13492
+
+iVBORw0KGgoAAAANSUhEUgAAAjEAAAD1CAYAAAChiaR/AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml]: We don't care about storing an actual image. You can just keep ~4 lines here.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml [hunk 3]
```diff
@@ -0,0 +1,294 @@
+Return-Path: <qnguyen@linagora.com>
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml]: Anonimation : 
 - Try avoid putting your real mail (spam...)
 - Don't internals of Linagora mail routing in here

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java [hunk 4]
```diff
@@ -106,10 +106,22 @@ public void appendMessageWithInlinedAttachmentToMailbox(String user, String mail
...
+    @Given("^\"([^\"]*)\" mailbox \"([^\"]*)\" contains a message \"([^\"]*)\" with multiple same inlined attachments \"([^\"]*)\"$")
+    public void appendMessageWithSameInlinedAttachmentsToMailbox(String user, String mailbox, String messageName, String attachmentId) throws Throwable {
+        MailboxPath mailboxPath = new MailboxPath(MailboxConstants.USER_NAMESPACE, user, mailbox);
+
+        mainStepdefs.jmapServer.serverProbe().appendMessage(user, mailboxPath,
+            ClassLoader.getSystemResourceAsStream("eml/sameInlinedImages.eml"), new Date(), false, new Flags());
+
+        attachmentsByMessageId.put(messageName, attachmentId);
+
+        blobIdByAttachmentId.put(attachmentId, "4000c5145f633410b80be368c44e1c394bff9437");
```
Comment (strokyl) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java]: Why don't you use the `ONE_ATTACHMENT_EML_ATTACHMENT_BLOB_ID` constant here.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java [hunk 5]
```diff
@@ -167,6 +167,11 @@ public void appendMessageBeginningByALongLine(String messageName, String mailbox
         appendMessage(messageName, "eml/longLine.eml");
     }
 
+    @Given("^the user has a message \"([^\"]*)\" in the \"([^\"]*)\" with multiple same inlined attachments \"([^\"]*)\"$")
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: "mailbox" missing

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+p1Spkc5kKFQa5HfuonbiTyhWCcsH14N0e+2WDWXB+Rfv0bXG4o0e4HjQ63bJvvoA/TM/o/nOcyTU
+DuVqGSORpNzoEo0nKdZbmDGD2vIcw933kU//mKE3HkW260HBAgyneOCVh7+p/UcJtFbh00q7POZp
+YHcuF5j4/AyRlRnUkyaG3o8fTROPxhASkp6N0i6Sb67QK9XRinBRBVUBRQWpQaRnTWQj3Wng5L8F
+oNfnjkiBQIVMEipl8D1AD05Lx0IpWaiAqoFUQWggVPAE9BRQ9IDahA5D/ZDqg2gCmc51Dn8cwD9J
+sPrq/TLkzB1SIqBEYWAHTF0DEQFeB7ze+hRBD3AVcOW6Yx96XfB7oAhIGfDZPbBrEsxBCKVBa84e
+smYfEVdkQHfXJnV/9TqMwJGUkAtBOg2tBnQc8Lrrtd8LnPl+0BakEpRiPQJmDlQJ0gDCAYMyBMK7
+8CnXLY8BZz8ZQPvvBzXdUz0VFDfoer4elNlEIqDLdz/CRA+EEmishECELveJXutyq8YPJBKaq2vO
+uzddEUCYMwdlEqS33u1k0GQ8Fbw6+CJwosh/DR5//eETnNOyIDPBfcS6LRW0zmvfAX7ziQCs99/b
+l5w+tG7lMgAxAEqPIJ8ueRMfK6mXVhkwhxYA+XB/HYS7+MI+9Uox8NQzyZ9PzbzwINs4zhYTP/3e
+0U38km3X+Aeuu0L8ysHGdwAAAABJRU5ErkJggg==
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml]: maybe I'm blind but I see this attachment only once.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java (hunk 4 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java]: Yes, it is and I changed

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java (hunk 5 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: I added

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml (hunk 6 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/eml/sameInlinedImages.eml]: Yes, It has only once attachment but it's display multiple times based on CID. Detail issue is here: https://github.com/linagora/james-project/issues/345
Basically, it's fixed by INBOX team so in James, we've just added new integration test for that input.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature [hunk 7]
```diff
@@ -201,10 +201,25 @@ Feature: GetMessages method
...
     And the id of the message is "m1"
+
+Scenario: Retrieving message which contains multiple same inlined attachments
+    Given the user has a message "m1" in the "inbox" with multiple same inlined attachments "ia1"
+    When the user ask for messages "m1"
+    Then no error is returned
+    And the list should contain 1 message
+    And the hasAttachment of the message is "true"
+    And the list of attachments of the message contains 1 attachments
+    And the first attachment is:
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature]: most values here are magical. Nothing from the test spec allows to specify these assertions.

Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature]: Maybe I don't understand your comment, could you explain me more?
Those value are getting from .eml file and it had function to assert those value 
```
private void assertAttachment(String attachment, DataTable attachmentProperties) {
        attachmentProperties.asList(TableRow.class)
            .stream()
            .forEach(entry -> assertThat(jsonPath.<Object>read(attachment + "." + entry.getKey())).isEqualTo(entry.getValue()));
    }
```

Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature]: the eml is an implementation detail : you should be able to understand the whole test by reading the gherkin file. Here, the values you require are not described earlier in the gherkin and I think that's a problem.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java [hunk 8]
```diff
@@ -167,7 +167,7 @@ public void appendMessageBeginningByALongLine(String messageName, String mailbox
         appendMessage(messageName, "eml/longLine.eml");
     }
 
-    @Given("^the user has a message \"([^\"]*)\" in the \"([^\"]*)\" with multiple same inlined attachments \"([^\"]*)\"$")
+    @Given("^the user has a message \"([^\"]*)\" in the mailbox \"([^\"]*)\" with multiple same inlined attachments \"([^\"]*)\"$")
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: sorry but usually mailbox is after the mailbox name

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature (hunk 7 above)
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature]: unfortunately that's not the first test where we do this. If think it's out of the scope of this issue to fix these test, unless you have a simple solution for this.

Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature]: the scenario is : Retrieving message which contains multiple same inlined attachments
It lacks what is expected : should return a single attachment

For such a scenario I expect to deliver an email in someone inbox, then retrieve the single attachment and check its cid.

I really think the remaining is not needed.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java (hunk 8 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: Yes

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature (hunk 7 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature]: I changed. Could you take a look?

**Comment by quynhn** [comment]
test this please

**Comment by rouazana** [comment]
You should rebase on master if you want this PR to build

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
Failed tests: 
  expected:<[tru]e> but was:<[fals]e>
  expected:<[tru]e> but was:<[fals]e>
```

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
Tests in error: 
  ESReporterTest.timeMetricsShouldBeReportedWhenJmapRequestsReceived » Connect C...
```

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
test this please

**Comment by rouazana** [comment]
test this please

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
org.apache.james.jmap.cassandra.CassandraUserProvisionningConcurrencyTest
09:00:19.225 [ERROR] c.d.d.c.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 1000 milliseconds
09:00:20.226 [ERROR] c.d.d.c.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 2000 milliseconds

Results :

Failed tests: 
  expected:<[tru]e> but was:<[fals]e>
  expected:<[tru]e> but was:<[fals]e>
```

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
Tests in error: 
  CassandraUidProviderTest.nextUidShouldGenerateUniqueValuesWhenParallelCalls:103 » Completion
```

**Comment by chibenwa** [comment]
test this please

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
org.apache.james.jmap.cassandra.CassandraUserProvisionningConcurrencyTest
02:10:16.654 [ERROR] c.d.d.c.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 1000 milliseconds
02:10:17.655 [ERROR] c.d.d.c.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 2000 milliseconds

Results :

Failed tests: 
  expected:<[tru]e> but was:<[fals]e>
  expected:<[tru]e> but was:<[fals]e>
```

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] org.junit.ComparisonFailure: expected:&lt;[tru]e&gt; but was:&lt;[fals]e&gt;
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] 	at org.apache.james.jmap.methods.integration.cucumber.GetMessagesMethodStepdefs.assertHasAttachmentOfTheFirstMessage(GetMessagesMethodStepdefs.java:415)
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] 	at ���.And the hasAttachment of the message is "true"(cucumber/GetMessages.feature:249)
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] 
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] Tests run: 292, Failures: 2, Errors: 0, Skipped: 1, Time elapsed: 67.087 sec &lt;&lt;&lt; FAILURE! - in org.apache.james.jmap.cassandra.cucumber.CassandraGetMessagesMethodTest
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] And the hasAttachment of the message is "true"(Scenario: Retrieving message which contains multiple same inlined attachments)  Time elapsed: 0.009 sec  &lt;&lt;&lt; FAILURE!
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] org.junit.ComparisonFailure: expected:&lt;[tru]e&gt; but was:&lt;[fals]e&gt;
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] 
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] Scenario: Retrieving message which contains multiple same inlined attachments  Time elapsed: 0.01 sec  &lt;&lt;&lt; FAILURE!
[Building Java 8 james-1872-19f3acfb-553e-443f-ab5c-79574e4f771d] org.junit.ComparisonFailure: expected:&lt;[tru]e&gt; but was:&lt;[fals]e&gt;
```

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
Tests in error: 
  CassandraSieveRepositoryTest>AbstractSieveRepositoryTest.setActiveScriptShouldWork:188 » ScriptNotFound
```

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
Tests in error: 
  TestDiscardProtocol.testRecord:80 »  Timed out wait for be notified that read ...
```

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
Results :

Failed tests: 
  expected:<[tru]e> but was:<[fals]e>
  expected:<[tru]e> but was:<[fals]e>
```

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature [hunk 9]
```diff
@@ -74,9 +74,14 @@ Feature: Download endpoint
...
+  Scenario: A user should have access to an inlined attachment
     Given "usera@domain.tld" is connected
     And "usera@domain.tld" mailbox "INBOX" contains a message "m2" with an inlined attachment "ia1"
     When "usera@domain.tld" downloads "ia1"
     Then the user should be authorized
+
+  Scenario: A user should have access to multiple same inlined attachments
+    Given "usera@domain.tld" is connected
+    And "usera@domain.tld" mailbox "INBOX" contains a message "m2" with multiple same inlined attachments "ia1"
+    When "usera@domain.tld" downloads "ia1"
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: Can you double this line ? To ensure we can download it two times?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java [hunk 10]
```diff
@@ -446,6 +450,12 @@ public void assertAttachmentsHasSize(int numberOfAttachments) throws Throwable {
         assertThat(jsonPath.<List<Object>>read(ATTACHMENTS)).hasSize(numberOfAttachments);
     }
 
+    @Then("^the list of attachments of the message contains (\\d+) attachments with cid \"([^\"]*)\"?$")
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: You don't need to have several attachments as you are matching only one CID.
So should be "contains 1 attachment"

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java [hunk 11]
```diff
@@ -59,7 +59,7 @@
 @ScenarioScoped
 public class DownloadStepdefs {
 
-    private static final String ONE_ATTACHMENT_EML_ATTACHEMENT_BLOB_ID = "4000c5145f633410b80be368c44e1c394bff9437";
+    private static final String ONE_ATTACHMENT_EML_ATTACHMENT_BLOB_ID = "4000c5145f633410b80be368c44e1c394bff9437";
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java]: Why this change ? (ATTACHMENT duplicated)

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java [hunk 12]
```diff
@@ -218,12 +218,17 @@ public void appendMessageWithNoPreview(String messageName, String mailbox) throw
...
+    }
+
     private void appendMessage(String messageName, String emlFileName) throws Exception {
         ZonedDateTime dateTime = ZonedDateTime.parse("2014-10-30T14:12:00Z");
         MessageId id = mainStepdefs.jmapServer.getProbe(MailboxProbeImpl.class).appendMessage(userStepdefs.lastConnectedUser,
-                new MailboxPath(MailboxConstants.USER_NAMESPACE, userStepdefs.lastConnectedUser, DefaultMailboxes.INBOX),
+            new MailboxPath(MailboxConstants.USER_NAMESPACE, userStepdefs.lastConnectedUser, DefaultMailboxes.INBOX),
                 ClassLoader.getSystemResourceAsStream(emlFileName),
-                Date.from(dateTime.toInstant()), false, new Flags())
+            Date.from(dateTime.toInstant()), false, new Flags())
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: Indent fail ?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature (hunk 9 above)
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: Up... Do a double download here to show me that it works

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java (hunk 11 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java]: It's just correct to right name /ATTACHEMENT/ATTACHMENT

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature (hunk 9 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: Yes, it had:

> When "usera@domain.tld" downloads "ia1"
    And "usera@domain.tld" downloads "ia1"
    Then the user should be authorized

Is that your meaning?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java (hunk 12 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: Done

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java (hunk 10 above)
Comment (quynhn) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java]: I changed

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature (hunk 9 above)
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadEndpoint.feature]: Oh yes it is. sorry

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
```
11:23:49.294 [ERROR] c.d.d.c.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 1000 milliseconds
11:23:50.294 [ERROR] c.d.d.c.ControlConnection - [Control connection] Cannot connect to any host, scheduling retry in 2000 milliseconds

Results :

Tests in error: 
  Unauthorized
  Unauthorized
```

**Comment by chibenwa** [comment]
Just merged
